testsrc/jstests/sorting-comparators.js

```diff
@@ -0,0 +1,99 @@
+load("testsrc/assert.js");
+
+// This is basically the default algorithm specified in ECMA 22.1.3.25.1.
+// We will use it to test the output
+function verifyCompare(x, y, fn) {
+  if (x === undefined && y === undefined) {
+    return 0;
+  }
+  if (x === undefined) {
+    return 1;
+  }
+  if (y === undefined) {
+    return -1;
+  }
+  if (fn !== undefined) {
+    return fn(x, y);
+  }
+  var xs = x.toString();
+  var ys = y.toString();
+  if (xs < ys) {
+    return -1;
+  }
+  if (xs > ys) {
+    return 1;
+  }
+  return 0;
+}
+
+function sortAndCompare(a, fn) {
+  a.sort(fn);
+  for (var i = 1; i < a.length; i++) {
+    var c = verifyCompare(a[i - 1], a[i], fn);
+    if (c > 0) {
+      print(i + ": Compared " + a[i-1] + " and " + a[i] + " result = " + c);
+    }
+    assertTrue(c <= 0);
+  }
+}
+
+sortAndCompare(["a", "b", "c"]);
+
+sortAndCompare(["a", "b", "c"],
+  function(x, y) {
+    if (x < y) {
+      return -1;
+    }
+    if (x > y) {
+      return 1;
+    }
+    return 0;
+  });
+
+
+sortAndCompare(["a", "b", "c"],
+  function(x, y) {
+    if (x < y) {
+      return 1;
+    }
+    if (x > y) {
+      return -1;
+    }
+    return 0;
+  });
+
+// Torture tests devised by @sainen and @bensummers
```

Comment:
Thanks, but I'd say that attribution isn't in necessary/useful here. It's just a test, plus comparator itself comes from underscore. (And it's sain*a*en. 🙂 )